django/middleware/csrf.py

```diff
@@ -105,19 +105,38 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
+    def _get_current_token(self, request):
+        try:
+            if settings.CSRF_USE_SESSIONS:
+                return _sanitize_token(request.session[settings.CSRF_COOKIE_NAME])
```

Comment:
We could rename `CSRF_COOKIE_NAME` as `CSRF_ENTRY_NAME` or `CSRF_ENTRY_KEY`, and then using it as cookie name if we're cookie based and session key if we're session based. This would be an improvement over the current code, but probably not over separate settings.